### PR TITLE
Rework Java & Java Compat README files

### PR DESCRIPTION
**Pull request by pepone**

### CODE REVIEW

File: java/README.md [hunk 1]
```diff
@@ -1,311 +1,98 @@
...
-[3]: https://gradle.org
-[4]: http://proguard.sourceforge.net
-[5]: https://commons.apache.org/proper/commons-compress/
+# Ice for Java
+
+[Getting started] | [Examples] | [Maven packages] | [Documentation] | [Building from source]
+
+The [Ice framework] provides everything you need to build networked applications, including RPC, pub/sub, server deployment, and more.
+
+Ice for Java is the Ice Java implementation.
```
Comment (bernardnormier) [inline java/README.md:7]: ?

Should it be?
Ice for Java is the Java implementation of the Ice framework. It includes the latest Slice-to-Java mapping introduced in Ice X.Y.

File: java/README.md [hunk 2]
```diff
@@ -1,311 +1,98 @@
...
+
+import com.zeroc.demos.Ice.minimal.Demo.*;
+
+public class Client
+{
+    public static void main(String[] args)
+    {
+        try(com.zeroc.Ice.Communicator communicator = com.zeroc.Ice.Util.initialize(args))
+        {
+            communicator.getProperties().setProperty(
```
Comment (bernardnormier) [inline java/README.md:37]: Why do we do that?

File: java/README.md [hunk 3]
```diff
@@ -1,311 +1,98 @@
...
+```java
+// Server implementation (Server.java)
+
+public class Server
+{
+    public static void main(String[] args)
+    {
+        try(com.zeroc.Ice.Communicator communicator = com.zeroc.Ice.Util.initialize(args))
+        {
+            communicator.getProperties().setProperty(
```
Comment (bernardnormier) [inline java/README.md:57]: Ditto

File: java/README.md [hunk 4]
```diff
@@ -1,311 +1,98 @@
...
+                "Ice.Default.Package",
+                "com.zeroc.demos.Ice.minimal");
+
+            // Install shutdown hook to (also) destroy communicator during JVM shutdown.
```
Comment (bernardnormier) [inline java/README.md:61]: Should be a shutdown, not a destroy.

**Review by externl — COMMENTED** [review]
I think it's also worth clarifying a bit better what Java compat is.

File: java/README.md (hunk 1 above)
Comment (bernardnormier) [reply java/README.md:7]: It was introduced in Ice 3.7.

File: java-compat/README.md [hunk 5]
```diff
@@ -1,252 +1,105 @@
...
-[2]: https://doc.zeroc.com/ice/3.7/release-notes/supported-platforms-for-ice-3-7-10
-[3]: https://gradle.org
-[4]: https://commons.apache.org/proper/commons-compress/
+# Ice for Java Compat
+
+[Getting started] | [Examples] | [Maven packages] | [Documentation] | [Building from source]
+
+The [Ice framework] provides everything you need to build networked applications, including RPC, pub/sub, server deployment, and more.
+
+Ice for Java Compat is the Ice Java implementation that is backward-compatible with prior Ice releases.
```
[the remark this answers is not in the document]
Comment (bernardnormier) [reply java-compat/README.md:7]: I think we should use "source compatible" instead, e.g.:

Ice for Java Compat is a Java implementation of the Ice framework that is largely source compatible with Ice for Java 3.6 and prior releases.

File: java/README.md (hunk 2 above)
Comment (pepone) [reply java/README.md:37]: We have this in the minimal demo, but we can get rid of it as this demo doesn't use classes or exceptions.

File: README.md [hunk 6]
```diff
@@ -84,7 +84,17 @@ languages.
 
 ## Building Ice from source
 
-[C++](cpp/BUILDING.md) | [C#](csharp/BUILDING.md) | [Java](java) | [JavaScript/TypeScript](js) | [MATLAB](matlab) | [Objective-C](objective-c) | [PHP](php) | [Python](python) | [Ruby](ruby) | [Swift](swift)
+- [C++](cpp/BUILDING.md)
```
Comment (bernardnormier) [inline README.md:87]: I prefer the previous layout.

File: java-compat/README.md [hunk 7]
```diff
@@ -1,252 +1,106 @@
...
+```java
+// Server implementation (Server.java)
+
+public class Server
+{
+    public static void main(String[] args)
+    {
+        try(Ice.Communicator communicator = Ice.Util.initialize(args))
+        {
+            // Install a shutdown hook to ensure the communicator gets shutdown when
```
Comment (bernardnormier) [inline java-compat/README.md:55]: get shut down

File: java/README.md (hunk 1 above)
Comment (pepone) [reply java/README.md]: Fixed as suggested

File: java/README.md (hunk 2 above)
Comment (pepone) [reply java/README.md]: Removed

File: java/README.md [hunk 8]
```diff
@@ -1,311 +1,90 @@
...
+```java
+// Server implementation (Server.java)
+
+public class Server
+{
+    public static void main(String[] args)
+    {
+        try(com.zeroc.Ice.Communicator communicator = com.zeroc.Ice.Util.initialize(args))
+        {
+            // Install a shutdown hook to ensure the communicator gets shutdown when
```
Comment (bernardnormier) [inline java/README.md:56]: gets shut down

File: java/README.md [hunk 9]
```diff
@@ -1,311 +1,90 @@
...
+```java
+// Client implementation (Client.java)
+
+import com.zeroc.demos.Ice.minimal.Demo.*;
+
+public class Client
+{
+    public static void main(String[] args)
+    {
+        try(com.zeroc.Ice.Communicator communicator = com.zeroc.Ice.Util.initialize(args))
```
Comment (bernardnormier) [inline java/README.md:37]: Can we add a `import com.zeroc.*` to make the names shorter?

**Review by bernardnormier — APPROVED** [review]

**Pull request merged**